dbms/src/Functions/FunctionsString.cpp

```diff
@@ -5022,52 +5027,410 @@ class FunctionPosition : public IFunction
-            throw Exception(
-                fmt::format("Illegal argument of function {}", getName()),
-                ErrorCodes::ILLEGAL_TYPE_OF_ARGUMENT);
+        size_t row_num = c0_col->size();
 
-        if (c0_col->size() != c1_col->size())
+        auto col_res = ColumnInt64::create(row_num, 0);
+        PaddedPODArray<Int64> & vec_res = col_res->getData();
+        vec_res.resize(row_num);
+
+        if (c0_const && c1_string)
+        {
+            const String & c0_str = c0_const->getValue<String>();
+            if unlikely (c0_str.empty())
+            {
+                for (size_t i = 0; i < row_num; i++)
+                    vec_res[i] = 1;
+            }
+            else
+            {
+                constVector(c0_const->getValue<String>(), c1_string->getChars(), c1_string->getOffsets(), vec_res);
+            }
+        }
+        else if (c0_string && c1_string)
+        {
+            vectorVector(
+                c0_string->getChars(),
+                c0_string->getOffsets(),
+                c1_string->getChars(),
+                c1_string->getOffsets(),
+                vec_res);
+        }
+        else if (c0_string && c1_const)
+        {
+            vectorConst(c0_string->getChars(), c0_string->getOffsets(), c1_const->getValue<String>(), vec_res);
+        }
+        else
             throw Exception(
-                fmt::format("Function {} column number is inconformity", getName()),
-                ErrorCodes::LOGICAL_ERROR);
+                "Illegal columns " + block.getByPosition(arguments[0]).column->getName() + " and "
+                    + block.getByPosition(arguments[1]).column->getName() + " of arguments of function " + getName(),
+                ErrorCodes::ILLEGAL_COLUMN);
 
-        auto col_res = ColumnInt64::create();
-        int val_num = c0_col->size();
-        col_res->reserve(val_num);
+        block.getByPosition(result).column = std::move(col_res);
+    }
+
+    static void vectorVector(
+        const ColumnString::Chars_t & col0_data,
+        const ColumnString::Offsets & col0_offsets,
+        const ColumnString::Chars_t & col1_data,
+        const ColumnString::Offsets & col1_offsets,
+        PaddedPODArray<Int64> & res)
+    {
+        size_t pos;
+        size_t row_num = col0_offsets.size();
+        ColumnString::Offset prev_col0_str_offset = 0;
+        ColumnString::Offset prev_col1_str_offset = 0;
 
-        for (int i = 0; i < val_num; i++)
+        for (size_t i = 0; i < row_num; i++)
         {
-            c0_col->get(i, c0_field);
-            c1_col->get(i, c1_field);
+            size_t col0_str_len = col0_offsets[i] - prev_col0_str_offset - 1;
+            size_t col1_str_len = col1_offsets[i] - prev_col1_str_offset - 1;
+
+            if unlikely (col0_str_len == 0)
+            {
+                res[i] = 1;
+            }
+            else
+            {
+                LibCASCIICaseSensitiveStringSearcher searcher = LibCASCIICaseSensitiveStringSearcher(
```

Comment:
> `position` should treat the input string as binary string, but `LibCASCIICaseSensitiveStringSearcher` actually use the `strstr` function, which only accept c style string, so I think it can not handle the case that there is an `\0` in the middle of the input string.

done, I replace it with `memmem`